app/src/main/java/com/github/se/polyfit/ui/components/AddIngredientPopup.kt

```diff
@@ -0,0 +1,197 @@
+package com.github.se.polyfit.ui.components
+
+import android.util.Log
+import androidx.compose.foundation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import com.github.se.polyfit.ui.theme.*
+import com.github.se.polyfit.ui.utils.removeLeadingZerosAndNonDigits
+
+// Constants
+// Add to list for new nutrition fields
+val NUTRITION_LABELS = listOf("Serving Size", "Calories", "Carbs", "Fat", "Protein")
+val NUTRITION_UNITS = listOf("g", "kcal", "g", "g", "g")
+// Temporary list of available ingredient to search
+val TMP_AVAILABLE_INGREDIENT = listOf("Apple", "Banana", "Carrot", "Date", "Eggplant", "apes")
+
+@Composable
+fun NutritionFacts() {
```

Comment:
This change would facilitate easier integration with the ViewModel
```suggestion
fun NutritionFacts(ingredients: List<Ingredient>) {
```

Reply:
Perhaps the naming of this component is not clear. The `NutritionFacts` is for editing the one single ingredient you are trying to add to the list of ingredient in a meal. For example, I want to add asparagus as one of the ingredients to my meal, I search up asparagus in the `SearchBar` and the app pulls from database and it says "asparagus has 15kcal". But I know that my asparagus is 10kcal, then I want to edit that at the field Calories in `NutritionFacts`. So I don't think it makes sense to pass in a list of ingredients.

Reply:
Yep, sorry I mistyped. I meant to say nutrients: List<Nutrient>, which should contains all the information you need for this class.

Reply:
I see, I think a view model of some sort should be passed in, as this field is for custom editing the ingredient's nutrition info if different from what was written in the database. Since we do not have a view model for it yet, I will add a todo comment for future integration